### PR TITLE
Add pysteps

### PR DESCRIPTION
**Pull request by aperezhortal**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by dnerini** [comment]
Yep, I confirm that I'm willing to be listed as maintainer.

File: recipes/pysteps/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: a2443ed3cd1636e60622a320cf7e7e8a0e857ad456e034ef19d084545ad1bc0e
+
+build:
+  number: 0
+  skip: True  # [py<36]"
```
Comment (chrisburr) [inline recipes/pysteps/meta.yaml]:
```suggestion
  skip: True  # [py<36]
```

File: recipes/pysteps/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  host:
+    - python
+    - numpy
+    - cython
+    - pip
+    - openmp 4.0.*  # [osx]
+    - attrdict
+    - jsmin
+    - scipy
+    - matplotlib
```
Comment (chrisburr) [inline recipes/pysteps/meta.yaml]: The `host` section should only contain dependencies which are needed to run `setup.py`.

Are these all needed or can some of them be removed? (`jsonschema`, `matplotlib`, `scipy` and probably some of the others)

File: recipes/pysteps/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  sha256: a2443ed3cd1636e60622a320cf7e7e8a0e857ad456e034ef19d084545ad1bc0e
+
+build:
+  number: 0
+  skip: True  # [py<36]"
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  build:
+    - numpy
```
Comment (chrisburr) [inline recipes/pysteps/meta.yaml]:
```suggestion
```

File: recipes/pysteps/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    - cython
```
Comment (chrisburr) [inline recipes/pysteps/meta.yaml]:
```suggestion
```

File: recipes/pysteps/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+test:
+  imports:
+    - pysteps
+
+about:
+  home: https://pysteps.github.io/
+  license: BSD-3-Clause
+  license_family: BSD
+  summary: 'Python framework for short-term ensemble prediction systems'
+  license_file:
```
Comment (chrisburr) [inline recipes/pysteps/meta.yaml]: In accordance with the terms of BSD-3-Clause, you need to specify a filename here. Ideally it should be included in the source tarball but if not you can include it in the recipe directory.

File: recipes/pysteps/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    - jsonschema
+  run:
+    - python
+    - openmp 4.0.*  # [osx]
+    - attrdict
+    - jsmin
+    - jsonschema
+    - matplotlib
+    - netCDF4
+    - numpy
```
Comment (chrisburr) [inline recipes/pysteps/meta.yaml]: Are you using the numpy C bindings? If yes, see [here](https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy):
```suggestion
    - {{ pin_compatible('numpy') }}
```
If not, you probably don't need numpy in the `host` section.

Comment (aperezhortal) [reply recipes/pysteps/meta.yaml]: Hi @chrisburr, yes, we have a Cython extension that needs numpy to run setup.py. Thanks for the fix.

File: recipes/pysteps/meta.yaml (hunk 2 above)
Comment (aperezhortal) [reply recipes/pysteps/meta.yaml]: Those dependencies are passed to setup() in setup.py, trough the *requirements* keyword. If we  don't include them conda-build gives the following error: 

`RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml`

File: recipes/pysteps/meta.yaml (hunk 5 above)
Comment (aperezhortal) [reply recipes/pysteps/meta.yaml]: Added in commit f4d208f , the LICENSE file is included in the sources.

File: recipes/pysteps/meta.yaml (hunk 4 above)
Comment (aperezhortal) [reply recipes/pysteps/meta.yaml]: Fixed in commit fc8d950

File: recipes/pysteps/meta.yaml (hunk 3 above)
Comment (aperezhortal) [reply recipes/pysteps/meta.yaml]: Fixed in commit fc8d950

**Comment by pulkkins** [comment]
I also confirm that I'm willing to be listed as a maintainer.

**Pull request merged**